tika-parsers/tika-parsers-standard/tika-parsers-standard-modules/tika-parser-cad-module/src/main/java/org/apache/tika/parser/dwg/DWGReadParser.java

```diff
@@ -0,0 +1,208 @@
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+
+            if (dwgc.isCleanDwgReadOutput()) {
+                // dwgread sometimes creates strings with invalid utf-8 sequences or invalid json (nan instead of NaN). replace them
+                // with empty string.
+
+                try (FileInputStream fis = new FileInputStream(tmpFileOut); FileOutputStream fos = new FileOutputStream(tmpFileOutCleaned)) {
+                    byte[] bytes = new byte[dwgc.getCleanDwgReadOutputBatchSize()];
+                    while (fis.read(bytes) != -1) {
+                        byte[] fixedBytes = new String(bytes, StandardCharsets.UTF_8)
+                            .replaceAll(dwgc.getCleanDwgReadRegexToReplace(), dwgc.getCleanDwgReadReplaceWith())
+                            .getBytes(StandardCharsets.UTF_8);
+                        fos.write(fixedBytes, 0, fixedBytes.length);
+                    }
+                } finally {
+                    FileUtils.deleteQuietly(tmpFileOut);
+                    tmpFileOut = tmpFileOutCleaned;
+                }
+
+            }
+            
+            // we can't guarantee the json output is correct so we try to ignore as many errors as we can
+            JsonFactory jfactory = JsonFactory.builder().enable(JsonReadFeature.ALLOW_MISSING_VALUES,JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS,JsonReadFeature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER).build();
+            JsonParser jParser = jfactory.createParser(tmpFileOut);
+            JsonToken nextToken = jParser.nextToken();
+            while ((nextToken = jParser.nextToken()) != JsonToken.END_OBJECT) {
+                if (nextToken == JsonToken.FIELD_NAME) {
+                    String nextFieldName = jParser.currentName();
+                    nextToken = jParser.nextToken();
+                    if (nextToken.isStructStart()) {
+
+                        if ("OBJECTS".equals(nextFieldName)) {
+                            // Start array
+                            jParser.nextToken();
+                            while (jParser.nextToken() != JsonToken.END_ARRAY) {
+                                parseDwgObject(jParser, (nextTextValue) -> {
+
+                                    try {
+										xhtml.characters(cleanupDwgString(nextTextValue));
+										xhtml.newline();
+									} catch (SAXException e) {
+										LOG.error("Could not write next text value {} to xhtml stream", nextTextValue);
+									}
+                                });
+                            }
+                        }  else if ("FILEHEADER".equals(nextFieldName)) {
+                            parseHeader(jParser,metadata);
+                        } else {
+                            jParser.skipChildren();
+                        }
+                    }
+                }
+            }
+            jParser.close();
+        } finally {
+            FileUtils.deleteQuietly(tmpFileOut);
```

Comment:
Should also delete tmpfileoutcleaned in case there's an exception?

Reply:
Added tmpFileOutCleaned to be deleted to make sure it is deleted if we encounter exception